connectors/activemq/src/main/java/forklift/producers/ActiveMQProducer.java

```diff
@@ -185,51 +204,48 @@ private Message prepAndValidate(ForkliftMessage message,
     private Message forkliftToJms(ForkliftMessage message) throws ProducerException {
 
         try {
-            Message msg = null;
-            if (message.getJmsMsg() != null ) {
-                try {
-                    msg = message.getJmsMsg();
-                } catch (Exception e) {
-                    throw new ProducerException("Error assigning Forklift JMS message to new MSG", e);
-                }
-            } else if (message.getMsg() != null ) {
-                try {
-                    msg = session.createTextMessage(message.getMsg());
-                } catch (Exception e) {
-                    throw new ProducerException("Error creating new jms TextMessage", e);
-                }
-            }
+            final Message msg = session.createTextMessage(message.getMsg());
+
+            // Set the jms correlation id.
+            if (message.getId() != null && !message.getId().trim().equals(""))
+                msg.setJMSCorrelationID(message.getId());
+
             setMessageHeaders(msg, message.getHeaders());
             setMessageProperties(msg, message.getProperties());
             return msg;
         } catch (Exception e) {
-            throw new ProducerException("Error creating JMS Message, Forklift message was null", e);
+            throw new ProducerException("Error creating JMS Message, or Forklift message was null", e);
```

Comment:
We might as well leave off the "Forklift message was null" bit. That should be fairly obvious when you get a NPE. If it really matters, I think it's better to check whether the message is null and throw just for that.